tests/console/suse_module_tools.pm

```diff
@@ -0,0 +1,57 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Testing functionality of modhash and modsign-verify commands.
+# A random module from available loadable modules on the system is picked
+# for testing purposes.
+# * modhash: the output is checked for correct format
+# * modsign-verify: the output is checked for correct format
+# Maintainer: Vasileios Anastasiadis <vasilios.anastasiadis@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+use version_utils 'is_sle';
+
+sub run {
+    select_console 'root-console';
+
+    # Find a random module available on the system
+    # Get a list of all modules
+    my $mds = script_output("find /lib/modules/\$(uname -r) -type f -name '*.ko'");
+    # Get the first module on the list
+    my $pth = (split '\n', $mds)[0];
+
+    # Test modhash command
+    if (!is_sle('=12-sp1') && !is_sle('=12-sp5')) {
+        # Get the output
+        my $hash = script_output("modhash $pth");
+        # Check for incorrect output
+        my $rgx = qr/$pth: [0-9a-fA-F]+/;
+        die('modhash failed') unless ($hash =~ $rgx);
+    }
+    # testing modsign-verify command
+    if (is_sle('<15-sp1')) {
+        # Get the output
+        my $out = script_output("modsign-verify $pth || :");
+        # Check for incorrect output
+        if (index($out, "good signature") == -1
```

Comment:
```suggestion
assert_script_run "modsign-verify $pth | grep 'good signature'";
```

Reply:
That was my first approach, but it is not enough - the module may not have a good signature - it may be unsigned or any of the other available possibilities listed, but modsign-verify still works since it detects that. It is only considered a failure if it outputs something else (I took all possible outputs from the modsign-verify code).

Reply:
You can still grep a regular expression - it will be one line instead of many and the assert_script_run is also more readable from screenshots so it is IMO easier to review as well.

Reply:
I followed your advice. Now the test runs in one grep session to find all possible correct outputs.